### PR TITLE
Friend notifcation improved

### PR DESCRIPTION
**Pull request by ThunderIW**
The system can do the following things

- User can send a friend Request to a particular user by using the addFriendModal which has been modified to be added to newly created fields in the database  so that friends are not forced  to be friends 
- Receiver can accept friend request putting the accepted friend  both the sender and receiver friendlist in the database 
- UI which can be seen below and is integrated into the main appshell 

image

image
 

**receiver end** 
image

**sender side**
image

after accepted friend request both sender and receiver 
image
image
new friend thunder90 added to Toby side

### CODE REVIEW

File: Application/Frontend/components/FriendsColumn/AddFriendModal.tsx
Comment (TobyNguyen710) [inline Application/Frontend/components/FriendsColumn/AddFriendModal.tsx:1]: Modified so does not forced adding to the friend list

File: Application/Frontend/components/FriendsColumn/FriendRequest.jsx
Comment (TobyNguyen710) [inline Application/Frontend/components/FriendsColumn/FriendRequest.jsx:1]: This is a component for friend request

File: Application/Frontend/components/Navbar/notifcation.tsx
Comment (TobyNguyen710) [inline Application/Frontend/components/Navbar/notifcation.tsx:1]: This is the notification component

File: Application/Frontend/package-lock.json
Comment (TobyNguyen710) [inline Application/Frontend/package-lock.json:1]: just a json change

File: Application/Frontend/package.json
Comment (TobyNguyen710) [inline Application/Frontend/package.json:1]: This is also a json change.

File: Application/Frontend/pages/TestSendFriendRequestPage.tsx
Comment (TobyNguyen710) [inline Application/Frontend/pages/TestSendFriendRequestPage.tsx:1]: This is a test page

File: Application/Frontend/pages/accord.tsx
Comment (TobyNguyen710) [inline Application/Frontend/pages/accord.tsx:1]: changes in the main accord page

File: Application/Frontend/pages/api/Get-SentFriendRequest.ts
Comment (TobyNguyen710) [inline Application/Frontend/pages/api/Get-SentFriendRequest.ts:1]: Api for get sent friend request

File: Application/Frontend/pages/api/Get-pendingFriendRequest.ts
Comment (TobyNguyen710) [inline Application/Frontend/pages/api/Get-pendingFriendRequest.ts:1]: Api for get pending friend request

File: Application/Frontend/pages/api/accept-friend-request.ts
Comment (TobyNguyen710) [inline Application/Frontend/pages/api/accept-friend-request.ts:1]: api for accept friend request

File: Application/Frontend/pages/api/add-friend.ts
Comment (TobyNguyen710) [inline Application/Frontend/pages/api/add-friend.ts:1]: modified add friend api

File: Application/Frontend/pages/api/add_freind_request_to_user.ts
Comment (TobyNguyen710) [inline Application/Frontend/pages/api/add_freind_request_to_user.ts:1]: api to add friend to users

File: Application/Frontend/pages/api/get-username-from-id.ts
Comment (TobyNguyen710) [inline Application/Frontend/pages/api/get-username-from-id.ts:1]: api to get username from id

File: Application/Frontend/pages/get_friendRequest.tsx
Comment (TobyNguyen710) [inline Application/Frontend/pages/get_friendRequest.tsx:1]: Just a test page

File: Application/Frontend/pages/notify.tsx
Comment (TobyNguyen710) [inline Application/Frontend/pages/notify.tsx:1]: empty file

File: package-lock.json
Comment (TobyNguyen710) [inline package-lock.json:1]: package update

File: package.json
Comment (TobyNguyen710) [inline package.json:1]: pakage update

**Review by TobyNguyen710 — APPROVED** [review]
The entire changes work as intended:
- Able to add friend without forcing to accept friend request
- Able to see friend request notification
- Able to accept the friend request and update the friend list.

File: Application/Frontend/components/FriendsColumn/AddFriendModal.tsx
Comment (ColinLefter) [inline Application/Frontend/components/FriendsColumn/AddFriendModal.tsx:5]: Great use of the built-in notification system from Mantine. Adding responsive notifications throughout the application is one of the next steps toward finalizing our application UX.

File: Application/Frontend/components/FriendsColumn/AddFriendModal.tsx
Comment (ColinLefter) [inline Application/Frontend/components/FriendsColumn/AddFriendModal.tsx:14]: This will cause problems if somebody intentionally added whitespace in their username and this type of check is not consistent throughout all other components of the application that rely on username checks. Instead of using `.trim` everywhere, we should ensure users cannot upload usernames with whitespace. However, this would be complicated as the changes in account data are done internally through the provider.

File: Application/Frontend/components/FriendsColumn/AddFriendModal.tsx
Comment (ColinLefter) [inline Application/Frontend/components/FriendsColumn/AddFriendModal.tsx:40]: Great distinction in notification colors. Good UX principles.

File: Application/Frontend/components/FriendsColumn/AddFriendModal.tsx
Comment (ColinLefter) [inline Application/Frontend/components/FriendsColumn/AddFriendModal.tsx:65]: Excellent use of our custom context that facilitates real-time updates. Great work in using this!

File: Application/Frontend/components/FriendsColumn/FriendRequest.jsx
Comment (ColinLefter) [inline Application/Frontend/components/FriendsColumn/FriendRequest.jsx:24]: Please update your endpoint names. There is a typo in this endpoint, and it is poorly formatted (please use hyphens instead).

File: Application/Frontend/components/FriendsColumn/FriendRequest.jsx
Comment (ColinLefter) [inline Application/Frontend/components/FriendsColumn/FriendRequest.jsx:46]: Please use Mantine components. This div can be replaced with a Mantine Stack.

File: Application/Frontend/components/FriendsColumn/FriendRequest.jsx
Comment (ColinLefter) [inline Application/Frontend/components/FriendsColumn/FriendRequest.jsx:57]: Use the button from Mantine. The reason why this is so important is because if I use the theme toggle, none of these HTML elements will be styled accordingly.

File: Application/Frontend/components/FriendsColumn/FriendRequest.jsx
Comment (ColinLefter) [inline Application/Frontend/components/FriendsColumn/FriendRequest.jsx:59]: You can condense this to just `message?`

File: Application/Frontend/components/Navbar/notifcation.tsx
Comment (ColinLefter) [inline Application/Frontend/components/Navbar/notifcation.tsx:21]: Again, please revise your API endpoint names to follow our naming standard. This endpoint should be renamed to `get-sent-friend-requests`

File: Application/Frontend/components/Navbar/notifcation.tsx
Comment (ColinLefter) [inline Application/Frontend/components/Navbar/notifcation.tsx:26]: Rename this API endpoint as well

File: Application/Frontend/components/Navbar/notifcation.tsx
Comment (ColinLefter) [inline Application/Frontend/components/Navbar/notifcation.tsx:54]: This is an example of a well-named API endpoint

File: Application/Frontend/components/Navbar/notifcation.tsx
Comment (ColinLefter) [inline Application/Frontend/components/Navbar/notifcation.tsx:113]: Mantine actually had some nice loading skeletons that you could use as well. However, a loading notification is ok for now.

File: Application/Frontend/pages/api/accept-friend-request.ts
Comment (ColinLefter) [inline Application/Frontend/pages/api/accept-friend-request.ts:29]: Remove these GPT comments to improve relevance and readability

File: Application/Frontend/pages/api/add_freind_request_to_user.ts
Comment (ColinLefter) [inline Application/Frontend/pages/api/add_freind_request_to_user.ts:19]: You should not be using this default is from MongoDB but rather our one. However, if this collection does not have a unique ID field and does not need one, you could keep this as long as it works.

File: Application/Frontend/pages/api/add_freind_request_to_user.ts
Comment (ColinLefter) [inline Application/Frontend/pages/api/add_freind_request_to_user.ts:15]: You also don't need to create a test collection for this particular case as you can use the Accounts collection directly. You can just create two accounts with the provider to test things that require two users. If you are testing fields that don't currently exist in any document in the collection, you can just add them to existing collections as long as they don't break anything or create the new fields as part of account syncing (in the registration endpoint)

**Review by ColinLefter — CHANGES_REQUESTED** [review]
Everything looks good, but just address the changes I mentioned in the comments. After that, this can be merged safely (assuming no additional bugs are added)

**Comment by ThunderIW** [comment]
Fixed all issues, removed test API class, and renamed my APIs that are in use with Hypens 
these included the 
- Get-SentPendingFriendList
- Get-RecivedPendingFriendList 

These are now renamed according 
- Get-Sent-Pending-Friend-List
- Get-Recived-Pending-Friend-List

File: Application/Frontend/pages/api/accept-friend-request.ts
Comment (ThunderIW) [reply Application/Frontend/pages/api/accept-friend-request.ts]: GPT Comments have been removed

**Review by ThunderIW — COMMENTED** [review]
Fixed all issues, removed test API class,renamed my APIs that are in use with Hypens, remove GPT lines in API
these included the

Get-SentPendingFriendList.ts
Get-RecivedPendingFriendList.ts
These are now renamed according

Get-Sent-Pending-Friend-List.ts
Get-Recived-Pending-Friend-List.ts

**Review by ColinLefter — APPROVED** [review]
Thank you for implementing the requested changes! Great work!

**Pull request merged**
